src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -758,10 +759,41 @@ private static bool IsAllPunctuation(string filterText)
         /// </summary>
         private static bool IsPotentialFilterCharacter(char c)
         {
-            // TODO(cyrusn): Actually use the right unicode categories here.
+            // TODO(cyrusn): Actually use the right Unicode categories here.
             return char.IsLetter(c)
                 || char.IsNumber(c)
                 || c == '_';
         }
+
+        /// <summary>
+        /// This comparer can handle expanded items.
+        /// </summary>
+        private class CompletionItemSortingComparer : IComparer<VSCompletionItem>
+        {
+            public static CompletionItemSortingComparer Instance { get; } = new CompletionItemSortingComparer();
+
+            public int Compare(VSCompletionItem x, VSCompletionItem y)
+            {
+                var xHasPrefix = HasExpandedPrefix(x);
+                var yHasPrefix = HasExpandedPrefix(y);
+
+                if (xHasPrefix == yHasPrefix)
+                {
+                    // Don't use Ordinal because we want lowercase to be "smaller" than uppercase.
+                    // e.g. "Goo" before "GoTo"
+                    return string.Compare(x.SortText, y.SortText, StringComparison.InvariantCulture);
```

Comment:
fascinating.  you added a test for ths?

Reply:
I reverted this to keep the order identical as before (Using OrdinalIgnoreCase)